js/article.js

```diff
@@ -1,49 +1,117 @@
+    this.readLength = (minuteRead <= 1) ? '1 minute' : minuteRead + ' minutes';
+    this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+    this.publishStatus = this.publishedOn ? 'published ' + this.daysAgo + ' days ago' : '(draft)';
+    this.body = marked(this.body);
+    return template(this);
+  };
 
-  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
-  this.publishStatus = this.publishedOn ? 'published ' + this.daysAgo + ' days ago' : '(draft)';
+//FUNCTIONALITY TO LOAD ALL ARTICLES IN DECENDING ORDER OF PUBLICATION
+  Article.loadAll = function(inputData) {
+    var articles = inputData.sort(function(a,b) {
+      return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+    }).map(function(ele) {
+      return new Article(ele);
+    });
+    Article.allArticles = articles;
+  };
 
-  return templateRender(this);
-};
+// THIS ADDS FUNCTIONALITY TO RETRIEVE DATA FROM LOCAL OR REMOTE SOURCE, WHICH PROCESS AND HANDS OFF CONTROL TO THE VIEW ********
+  Article.fetchAll = function(nextFunction) {
+    //if (localStorage.blogArticles) {
+    $.ajax({
+      type: 'HEAD',
+      url: 'data/blogArticles.json',
+      success: function(data, message, xhr) {
+        var eTag = xhr.getResponseHeader('eTag');
+        if (!localStorage.eTag || eTag !== localStorage.eTag) {
+          Article.getAll(nextFunction); // DONE: pass 'nextFunction' into Article.getAll();
+        } else {
+          Article.loadAll(JSON.parse(localStorage.blogArticles));
+          // Replace the following line with 'nextFunction' and invoke.
+          nextFunction();
+        }
+      }
+    });
+  };
 
-//FUNCTIONALITY TO LOAD ALL ARTICLES IN DECENDING ORDER OF PUBLICATION
-Article.loadAll = function(inputData) {
-  inputData.sort(function(a,b) {
-    return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
-  }).forEach(function(ele) {
-    Article.allArticles.push(new Article(ele));
-  });
-};
-
-// THIS ADDS FUNCTIONALITY TO RETRIEVE DATA FROM LOCAL OR REMOTE SOURCE, WHICH PROCESS AND HANDS OFF CONTROL TO THE VIEW
-Article.fetchAll = function() {
-
-  if (localStorage.hackerIpsum) {
-    Article.loadAll(JSON.parse(localStorage.hackerIpsum));
-    articleView.renderIndexPage();
-  } else {
-    $.getJSON('data/hackerIpsum.json', function(data, status, XHR) {
-      localStorage.hackerIpsum = JSON.stringify(data);
-      console.log(data);
-      Article.loadAll(data);
-      articleView.renderIndexPage();
+  Article.getAll = function(nextFunction) {
+    $.getJSON('data/blogArticles.json', function(responseData, message, xhr) {
+      localStorage.eTag = xhr.getResponseHeader('eTag');
+      Article.loadAll(responseData);
+      localStorage.blogArticles = JSON.stringify(responseData);
+      // Invoke our parameter.
+      nextFunction();
     });
   };
-};
+
+  // GETS # OF WORDS IN GIVEN ARTICLE
```

Comment:
I'm not positive, but I don't think you need this getNumWords function. It seems as if you are getting the number of words below in the NumWordsAll function. In class, we used this getNumWords so that we could assign the number of words each author has. You may want to comment it out for now if you plan on adding it that feature later.